mina-daemon/scripts/s3-blocks-uploader.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env sh
+
+if [ -z "$SOURCE_DIR" ]; then
+    echo "The SOURCE_DIR environment variable is not set or is empty."
+    exit 1
+fi
+
+if [ -z "$S3_BUCKET" ]; then
+    echo "The S3_BUCKET environment variable is not set or is empty."
+    exit 1
+fi
+
+if [ -z "$UPLOAD_INTERVAL" ]; then
+    echo "The UPLOAD_INTERVAL environment variable is not set or is empty."
+    exit 1
+fi
+
+if [ ! -z "$PREFIX" ]; then
+    S3_BUCKET="${S3_BUCKET}/${PREFIX}"
+    echo "Destination bucket is ${S3_BUCKET}"
+fi
+
+cd "$SOURCE_DIR"
+
+while true; do
+    # Loop through local files
+    echo "Checking for new blocks in $SOURCE_DIR"
+
+    blocks=$(ls | grep json)
+
+    # Check if there are any matching files
+    if [ -z "$blocks" ]; then
+        echo "No new blocks found in $SOURCE_DIR"
+    else
+        for local_file in $blocks; do
+            # Extract the filename without the path
+            block_name=$(basename "${local_file}")
+
+            # Check if the file exists in the S3 bucket
+            echo "Check if file ${block_name} exists in s3://${S3_BUCKET}"
+            if aws s3 ls "s3://${S3_BUCKET}/${block_name}" 2>&1; then
+                echo "=> File ${block_name} already exists in S3. Skipping."
+            else
```

Comment:
I can't believe that aws s3 does not have a flag like gsutil have `-n`. Because essentially you're making 2 api calls per file

Reply:
Yes, I agree. I'll keep that PR open and will double check it

Reply:
I checked it, there seem to be nothing on `s3` api. Also `sync` api seems to be useless as with many objects it will become very slow very quick.
However I saw there's a flag `--if-none-match` on `s3api` api 
Also I see this very old thread that has been opened/closed/opened for years
https://github.com/aws/aws-cli/issues/2874